### DISCUSSION TITLE
Integrating Dynamic Custom Layers into pgRouting for Custom Waypoints

### DISCUSSION THREAD

**Discussion by 3zzy** [pgRouting/pgrouting#2596] [Q&A]
Hello pgRouting Community,

I am currently working on a project that involves creating a navigation system with custom waypoints. My objective is to enable users to add their own points, lines, and areas on a map and utilize these custom elements for navigation. For example, users could drop pins at specific locations (like near the Eiffel Tower garden and at exit 3 of Bir-Hakim metro station) to set them as the source and destination. Additionally, I aim to include custom lines drawn by users as waypoints in their routes.

How can I effectively integrate dynamic layer data, such as these custom points, lines, and areas, into pgRouting? Specifically, I need to ensure that the pgRouting algorithm dynamically considers these custom layers when calculating routes. I am also interested in any examples, case studies, or documentation that detail such an integration. 

Thank you in advance for your assistance.

**Comment by cvvergara** [comment]
Have you tried [pgr_withPoints](https://docs.pgrouting.org/latest/en/pgr_withPoints.html) ?
